pkg/controller/jenkins/configuration/base/validate.go

```diff
@@ -60,6 +61,46 @@ func (r *ReconcileJenkinsBaseConfiguration) Validate(jenkins *v1alpha2.Jenkins)
 	return true, nil
 }
 
+func (r *ReconcileJenkinsBaseConfiguration) validateImagePullSecrets() (bool, error) {
+	var err error
+	valid := true
+	ips := r.jenkins.Spec.Master.ImagePullSecrets
+	for _, sr := range ips {
+		valid, err = r.validateImagePullSecret(sr.Name)
+		if err != nil || !valid {
+			return valid, err
+		}
+	}
+
+	return valid, err
+}
+
+func (r *ReconcileJenkinsBaseConfiguration) validateImagePullSecret(name string) (bool, error) {
+	secret := &corev1.Secret{}
+	err := r.k8sClient.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: r.jenkins.ObjectMeta.Namespace}, secret)
+	if err != nil && apierrors.IsNotFound(err) {
+		r.logger.V(log.VWarn).Info(fmt.Sprintf("Secret '%s' not found", name))
```

Comment:
improve error message `Secret %s not found defined in spec.master.imagePullSecrets`